src/Faithlife.Analyzers/AvailableWorkStateAnalyzer.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Faithlife.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public sealed class AvailableWorkStateAnalyzer : DiagnosticAnalyzer
+	{
+		public override void Initialize(AnalysisContext context)
+		{
+			context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.SimpleMemberAccessExpression);
+		}
+
+		public const string DiagnosticId = "FL0008";
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(s_rule); } }
+
+		private static void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+		{
+			var syntax = (MemberAccessExpressionSyntax)context.Node;
+
+			var iworkState = context.SemanticModel.Compilation.GetTypeByMetadataName("Libronix.Utility.Threading.IWorkState");
```

Comment:
In `OverloadWithStringComparerAnalyzer`, I call `RegisterCompilationStartAction` and perform `Compilation.GetTypeByMetadataName` just once in that method; if successful, I register a second callback:

https://github.com/Faithlife/FaithlifeAnalyzers/blob/ddcee4314be3e93dbc24d0cb41f602fb194d1763/src/Faithlife.Analyzers/OverloadWithStringComparerAnalyzer.cs#L14-L19

I believe I borrowed this pattern from some other Roslyn analyzer I found on the Internet... but I assume that the benefit is that the metadata lookup is performed just once when compilation starts, not on every single member access expression. 

I don't know what the performance implications are, but I imagine that it could add up over hundreds of member accesses? To fix this, we would need to refactor `CurrentAsyncWorkItemAnalyzer` and `UntilCanceledAnalyzer`. That could wait for a separate refactor after this is merged.